strr-api/src/strr_api/services/gcp_storage_service.py

```diff
@@ -0,0 +1,88 @@
+# Copyright © 2024 Province of British Columbia
+#
+# Licensed under the BSD 3 Clause License, (the "License");
+# you may not use this file except in compliance with the License.
+# The template for the license can be found here
+#    https://opensource.org/license/bsd-3-clause/
+#
+# Redistribution and use in source and binary forms,
+# with or without modification, are permitted provided that the
+# following conditions are met:
+#
+# 1. Redistributions of source code must retain the above copyright notice,
+#    this list of conditions and the following disclaimer.
+#
+# 2. Redistributions in binary form must reproduce the above copyright notice,
+#    this list of conditions and the following disclaimer in the documentation
+#    and/or other materials provided with the distribution.
+#
+# 3. Neither the name of the copyright holder nor the names of its contributors
+#    may be used to endorse or promote products derived from this software
+#    without specific prior written permission.
+#
+# THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS “AS IS”
+# AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO,
+# THE IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE
+# ARE DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE
+# LIABLE FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR
+# CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF
+# SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS
+# INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN
+# CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE)
+# ARISING IN ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE
+# POSSIBILITY OF SUCH DAMAGE.
+"""Manages Auth service interactions."""
+import uuid
+
+from flask import current_app
+from google.cloud import storage
+from google.oauth2 import service_account
+
+
+class GCPStorageService:
+    """Service to save and load files from gcp buckets."""
+
+    @classmethod
+    def registration_documents_bucket(cls):
+        """Get gcp bucket for saving or deleting registration documents."""
+
+        # project_id = current_app.config.get("GCP_CS_PROJECT_ID")
+        scope = current_app.config.get("GCP_CS_SA_SCOPE")
+        bucket_id = current_app.config.get("GCP_CS_BUCKET_ID")
+        # auth_key = current_app.config.get("GCP_AUTH_KEY")
+
+        # TODO: fix service account setup
+        service_account_info = {
```

Comment:
@pwei1018 using ppr-api as an example here, it looks like we need a few more environment variables for the gcp bucket connection:  https://github.com/bcgov/ppr/blob/main/ppr-api/src/ppr_api/callback/auth/token_service.py#L48